Add blog post: foremanctl 1.0 release announcement (#2247)
theforeman/theforeman.org: _posts/2025-10-06-foremanctl-10---deploying-foreman-in-containers-reaches-first-milestone.md

---
layout: post
title: foremanctl 1.0 - deploying Foreman in Containers reaches first milestone
date: 2025-10-06  7:47:50
author: Archana Kumari
tags:
- foreman
- foremanctl
- containers
- quadlet
- podman
- systemd
- release
- katello
- pulp
---

Over the past few months, we’ve been working on a new way to deploy and manage Foreman and its components (like Katello, Candlepin, and Pulp) using **containers**.
Today, we’re happy to share that **foremanctl 1.0** has been tagged, marking our first milestone toward a fully containerized Foreman experience.

<!--more-->

### What You Can Do with 1.0

With this first release, you can already:

- Set up a basic Foreman + Katello environment using containers.
- Use an external database configuration.
- Manage services with systemd and Quadlet, making it easier to start, stop, and maintain the setup.

Under the hood, `foremanctl` generates **Podman Quadlet** units that integrate directly with **systemd**, providing a consistent and declarative way to run Foreman services as containers — without relying on traditional installer RPMs.

> If you’d like to try it out, follow the updated [Quickstart Guide for 3.16](https://theforeman.org/manuals/3.16/quickstart_guide.html), which now includes step-by-step instructions for installing and running `foremanctl`. For detailed changes and improvements, see the [1.0.0 release notes](https://github.com/theforeman/foremanctl/releases/tag/1.0.0).

Reviewer: ```suggestion
If you’d like to try it out, follow the updated [Quickstart Guide for 3.16](https://docs.theforeman.org/3.16/Quickstart/index-katello.html), which now includes step-by-step instructions for installing and running `foremanctl`. For detailed changes and improvements, see the [1.0.0 release notes](https://github.com/theforeman/foremanctl/releases/tag/1.0.0).
```

---

### What’s Still in Progress

This release is an important first step, but it’s not yet a full replacement for the traditional installer setup. We’re continuing to work on:

- Containerizing Smart Proxy  
- Adding support for managing plugins (currently static list)  
- Improving SELinux and networking integration  
- Extending system requirement checks and configuration flexibility  

> **Note:** Smart Proxy services (like REX, Ansible, and DNS/DHCP) are not yet available in the release.

You can follow progress and upcoming work items in the [foremanctl milestone 2.0 roadmap](https://github.com/theforeman/foremanctl/milestone/2).

> Upcoming releases will build on this foundation: **3.16** as an *alpha milestone*, **3.17** as *beta*, and **3.18** targeting a *GA-ready containerized deployment flow*.

Reviewer: Not sure what the convention on promising features in future releases are but I'm wondering if the wording here should be more careful: Rather than state with certainty that these releases will represent these milestones, perhaps use wording like "is expected", "is planned", "should", etc.? Or just add a statement that plans can change?

Reviewer 2: I'd tone it down to a desire. Perhaps use `aim` instead of `targeting`?

---

### Try It Out and Share Feedback

> We’d love for you to experiment with `foremanctl`, try out the containerized setup, and tell us what works (and what doesn’t). Your feedback helps us smooth the rough edges and shape the next iterations.

Reviewer 3: ```suggestion
We’d love for you to experiment with [foremanctl](https://github.com/theforeman/foremanctl), try out the containerized setup, and tell us what works (and what doesn’t). Your feedback helps us smooth the rough edges and shape the next iterations.
```

> We’d love for you to experiment with [foremanctl](https://github.com/theforeman/foremanctl), try out the containerized setup, and tell us what works (and what doesn’t). Your feedback helps us smooth the rough edges and shape the next iterations.

Reviewer: Can you add a link to where users can share feedback from here?

Author: I assume that I've added the matrix link here, so do you think that would be good here?

Reviewer: Exactly :) I think that when users are asked to provide feedback, it's good to follow up on that with clear instructions on where and how to provide that feedback.

What I think could also work would be something like this:

```suggestion
We’d love for you to experiment with [foremanctl](https://github.com/theforeman/foremanctl), try out the containerized setup, and tell us what works (and what doesn’t). Your feedback helps us smooth the rough edges and shape the next iterations. To share your feedback, you can:

* Talk to us on Matrix (link to the channel)
* Open a thread in the [Foreman Community Forum](https://community.theforeman.org/) (<-- This would basically mean moving line 62 to here)
* Open an issue in the foremanctl repo (link to repo)
```

Reviewer 2: In the past we've opened feedback topic before to keep them central. Then in the announcement we linked to that post. A good example of that is https://community.theforeman.org/t/foreman-3-16-release-candidate-feedback/44135. You can see we linked to that in our 3.16 RC announcements (https://community.theforeman.org/t/foreman-3-16-0-rc1-is-now-ready-for-testing/44136 and https://community.theforeman.org/t/foreman-3-16-0-rc2-is-now-ready-for-testing/44221).

Author: @ekohl I can create the feedback thread on Discourse once the blog post is merged, so we can link both together

Reviewer 4: Don't blogposts get automatically a comment thread created on discourse? In the "blog" section: https://community.theforeman.org/c/blog/17

Not saying we should not have one dedicated for feedback in this case.

Reviewer 2: > @ekohl I can create the feedback thread on Discourse once the blog post is merged, so we can link both together

If you want to link to it, I'd create the general feedback first. Otherwise you need to create a PR after the fact to add the link.

> Don't blogposts get automatically a comment thread created on discourse? In the "blog" section: https://community.theforeman.org/c/blog/17

That's true. Question is, do we want to use a blog as a generic containerization feedback post?

Reviewer 4: Probably not, no.
But we should post a comment in the auto-created one, that people should look/post elsewhere.

Author: > If you want to link to it, I'd create the general feedback first. Otherwise you need to create a PR after the fact to add the link.

@ekohl yeah, that make sense! We’ll need a dedicated foremanctl 1.0 feedback thread anyway.

Author: > Don't blogposts get automatically a comment thread created on discourse? In the "blog" section: https://community.theforeman.org/c/blog/17

@evgeni @ekohl I don't see it created after the blog is already published? (i was thinking of linking original blog there)

Reviewer 4: If you go to https://theforeman.org/2025/10/foremanctl-10-deploying-foreman-in-containers-reaches-first-milestone.html and scroll down, there is a "start discussion" link that goes to https://community.theforeman.org/t/foremanctl-1-0-deploying-foreman-in-containers-reaches-first-milestone/44686

I think that post is hidden by default in the normal forum view.

Author: Hmm I see, thanks!

---
layout: post
title: foremanctl 1.0 - deploying Foreman in Containers reaches first milestone

> date: 2025-10-06  7:47:50

Reviewer 4: ```suggestion
date: 2025-10-17  6:20:50
```
